src/hooks/useSSO.ts

```diff
@@ -0,0 +1,24 @@
+import { useCallback } from "react";
+import firebase from "firebase/app";
+
+import { SSO_PROVIDER_ID } from "secrets";
+
+export const useSSO = () => {
+  const hasSSOProvider = SSO_PROVIDER_ID !== undefined;
+
+  const signInSSO = useCallback(() => {
+    if (!SSO_PROVIDER_ID) return;
+
+    const SSOProvider = new firebase.auth.SAMLAuthProvider(SSO_PROVIDER_ID);
```

Comment:
This hook is written generically as 'SSO', yet this implementation is specifically SAML related. There are other forms of SSO that can be implemented. What're the thoughts/plan/etc here?

Reply:
Changed the name in https://github.com/sparkletown/sparkle/pull/1546/commits/62de340d0e1eda6f11434e9fa870896e6698c0f7